src/Control/Applicative/Free/Fast.hs

```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE CPP                #-}
+{-# LANGUAGE GADTs              #-}
+{-# LANGUAGE RankNTypes         #-}
+#if __GLASGOW_HASKELL__ >= 707
+{-# LANGUAGE DeriveDataTypeable #-}
+#endif
+{-# OPTIONS_GHC -Wall #-}
+
+#ifndef MIN_VERSION_base
+#define MIN_VERSION_base(x,y,z) 1
+#endif
+--------------------------------------------------------------------------------
+-- |
+-- A faster free applicative.
+-- Based on <https://www.eyrie.org/~zednenem/2013/05/27/freeapp Dave Menendez's work>.
+--------------------------------------------------------------------------------
+module Control.Applicative.Free.Fast
+  ( ASeq(..)
+  , reduceASeq
```

Comment:
`ASeq` is an implementation detail necessary to implement `Ap`, is it not? If so, it should be moved to the bottom of the export list in its own Haddock section so that this is clear to the reader.